jiant/proj/main/modeling/primary.py

```diff
@@ -85,3 +89,31 @@ def wrap_jiant_forward(
     if is_multi_gpu and compute_loss:
         model_output.loss = model_output.loss.mean()
     return model_output
+
+
+class JiantTransformersModelFactory:
+    @staticmethod
+    def create_model(hf_model):
+        model_arch = ModelArchitectures.from_model_type(hf_model.config.model_type)
+        if model_arch == ModelArchitectures.BERT:
+            return JiantBERTModelWrapper(hf_model)
+
+
+class JiantTransformersModelWrapper:
+    def __init__(self, baseObject):
+        self.__class__ = type(
+            baseObject.__class__.__name__, (self.__class__, baseObject.__class__), {}
+        )
+        self.__dict__ = baseObject.__dict__
+
+
+class JiantBERTModelWrapper(JiantTransformersModelWrapper):
+    def __init__(self, baseObject):
+        super().__init__(baseObject)
+        self.hf_pretrained_encoder_with_pretrained_head = transformers.BertForPreTraining
+
+    def get_hidden_size(self):
+        return self.config.hidden_size
+
+    def get_hidden_dropout_prob(self):
+        return self.config.hidden_dropout_prob
```

Comment:
Can you add some docstrings to these? Their purpose isn't immediately clear.

Reply:
Can do. Just wanted to run this idea by you. This basically wraps Transformer models so that any inconsistencies can be abstracted from the model. For example, `hidden_size` has different member variable names in BERT and BART.

Reply:
I think abstracting the hidden_size/dropout prob is headed in the right direction. One thing I'd like to throw in is that it might not be worth making a different class for each model type. Can discuss more tomorrow.